--- src/global.cc       2022-03-17 23:39:26.810557474 -0700
+++ src/global.cc       2022-03-17 23:41:20.991885448 -0700
@@ -94,6 +94,11 @@
 #include <stdio.h>
 #include <stdarg.h>
 
+#if defined(__APPLE__)
+#define _DARWIN_C_SOURCE
+#endif
+#include <dlfcn.h>
+
 #ifdef HAVE_LIBMICROPYTHON
 std::string python_console;
 #endif
@@ -2793,7 +2798,20 @@
       }
 #endif // GNUWINCE
 #else
-      s=giac_aide_location;
+      Dl_info info;
+      if (dladdr((void*)protected_read_config, &info)) {
+        char const * soname = info.dli_fname;
+        s = string(soname);
+        size_t last_slash = s.rfind('/');
+        s = s.substr(0, last_slash+1);
+        if (s.size() > 0) {
+          s += "../share/giac/aide_cas";
+        } else {
+          s=giac_aide_location;
+        }
+      } else {
+        s=giac_aide_location;
+      }
       s=s.substr(0,s.size()-8);
 #endif
       if (s.size())
@@ -2837,7 +2855,21 @@
 #ifdef WIN32
     return "/cygdrive/c/xcas/";
 #endif
-    string s(giac_aide_location); // ".../aide_cas"
+    string s; // ".../aide_cas"
+    Dl_info info;
+    if (dladdr((void*)giac_aide_dir, &info)) {
+      char const * soname = info.dli_fname;
+      s = string(soname);
+      size_t last_slash = s.rfind('/');
+      s = s.substr(0, last_slash+1);
+      if (s.size() > 0) {
+        s += "../share/giac/aide_cas";
+      } else {
+        s=giac_aide_location;
+      }
+    } else {
+      s=giac_aide_location;
+    }
     // test if aide_cas is there, if not test at xcasroot() return ""
     if (!access(s.c_str(),R_OK)){
       s=s.substr(0,s.size()-8);
